pypeit/core/flexure.py

```diff
@@ -697,3 +704,520 @@ def calculate_image_offset(image, im_ref, nfit=3):
+        plot (bool, optional): Defaults to False.
+        noff (int, optional): Range in Ang to analyze labout emission line. Defaults to 5.
+        nfit_min (int, optional): Minimum number of pixels required to do a fit. Defaults to 20.
+
+    Returns:
+        tuple: np.ndarray's -- sky line wavelength, wavelength offset, 
+            error in wavelength offset, sky line width,
+            error in sky line width
+    """
+
+
+    dwave = []
+    diff = []
+    diff_err  = []
+    los = []
+    los_err= []
+    # Loop on known sky lines
+    for line in sky_waves: 
+        wline = [line-noff,line+noff] 
+        mw    = (wave > wline[0]) & (wave < wline[1])
+        
+        # Reuire minimum number
+        if np.sum(mw) <= nfit_min:
+            continue
+
+        p=[0,0,0,0]
+        # Guess
+        p0 = list(fitting.guess_gauss(wave[mw],flux[mw]))
+        # Fit
+        try:
+            p, pcov = fitting.fit_gauss(wave[mw],flux[mw], 
+                                w_out = 1./np.sqrt(ivar[mw]), 
+                                guesses=p0, nparam=4)
+            perr = np.sqrt(np.diag(pcov))
+        except:
+            p=p0
+            p[2] = -99
+            perr=p0
+        # Continue
+        d = p[2] - line
+
+        # For debugging
+        if plot:
+            gfit = fitting.gauss_4deg(wave[mw],*p)
+            plt.figure(figsize=(8,3)) 
+            plt.plot(wave[mw],gfit,'g')
+            plt.plot(wave[mw],flux[mw])
+            plt.title('{} {:0.2f} diff= {:0.3f}'.format(line,p[3],d))
+            plt.show()
+
+        # Check
+        if not np.isfinite(perr[2]):
+            perr[2] = 1000.
+        # Save
+        dwave = np.append(dwave,line)
+        diff = np.append(diff,d)
+        diff_err = np.append(diff_err,perr[2])
+        los = np.append(los,p[3])
+        los_err = np.append(los_err,perr[3])
+
+    # Cut on quality
+    m=(diff_err < 0.1) & (diff_err > 0.0)
+    # Return
+    return dwave[m],diff[m],diff_err[m],los[m],los_err[m]
+
+class MultiDetFlexure(DataContainer):
+    # Class to perform Multi-Detector flexure analysis
+    # Based on codes written by Marla Geha for DEIMOS
+
+    # Set the version of this class
+    version = '1.0.0'
+
+    datamodel = {'s1dfile': dict(otype=str, descr='spec1d filename'), 
+                 'PYP_SPEC': dict(otype=str, descr='PypeIt spectrograph name'),
+                 'ndet': dict(otype=int, descr='Number of detectors per spectrum'),
+                 'nslits': dict(otype=int, descr='Number of slits'),
+                 'spat': dict(otype=np.ndarray, atype=np.floating, descr='?? (ndet, nslits)'),
+                 'det': dict(otype=np.ndarray, atype=np.integer, descr='Detector number (ndet, nslits)'),
+                 'SN': dict(otype=np.ndarray, atype=np.floating, descr='S/N (ndet, nslits)'),
+                 'xpos': dict(otype=np.ndarray, atype=np.floating, descr='Slit x position [pixels] (nslits)'),
```

Comment:
Please change these tags to spec and spat. x and y do not exist.

Reply:
done